src/Camera_TEST.cc

```diff
@@ -156,6 +156,65 @@ TEST(Camera_TEST, CreateCamera)
   EXPECT_TRUE(badCam == nullptr);
 }
 
+/////////////////////////////////////////////////
+TEST(Camera_TEST, Topic)
+{
+  const std::string type = "camera";
+  const std::string name = "TestCamera";
+  const double updateRate = 30;
+  const bool alwaysOn = 1;
+  const bool visualize = 1;
+
+  // Factory
+  ignition::sensors::Manager mgr;
+
+  // Default topic
+  {
+    const std::string topic;
+    auto cameraSdf = CameraToSdf(type, name, updateRate, topic, alwaysOn,
+        visualize);
+
+    auto sensorId = mgr.CreateSensor(cameraSdf);
+    EXPECT_NE(ignition::sensors::NO_SENSOR, sensorId);
+
+    auto sensor = mgr.Sensor(sensorId);
+    EXPECT_NE(nullptr, sensor);
+
+    auto camera = dynamic_cast<ignition::sensors::CameraSensor *>(sensor);
+    ASSERT_NE(nullptr, camera);
+
+    EXPECT_EQ("/camera", camera->Topic());
+  }
+
+  // Convert to valid topic
+  {
+    const std::string topic = "/topic with  spaces/@~characters//";
+    auto cameraSdf = CameraToSdf(type, name, updateRate, topic, alwaysOn,
+        visualize);
+
+    auto sensorId = mgr.CreateSensor(cameraSdf);
+    EXPECT_NE(ignition::sensors::NO_SENSOR, sensorId);
+
+    auto sensor = mgr.Sensor(sensorId);
+    EXPECT_NE(nullptr, sensor);
+
+    auto camera = dynamic_cast<ignition::sensors::CameraSensor *>(sensor);
+    ASSERT_NE(nullptr, camera);
+
+    EXPECT_EQ("/topic_with__spaces/characters", camera->Topic());
```

Comment:
hmm I'm getting test failure here

```
Expected equality of these values:
  "/topic_with__spaces/characters"
  camera->Topic()
    Which is: "/topic_with_spaces/characters"
```

notice that there is only 1 `_` after `with` in the returned topic name. Do you get this failure?

Reply:
I originally created this PR against Dome and backported to Blueprint without testing locally. After digging a bit, I noticed this is a difference between `tinyxml` and `tinyxml2`: https://github.com/osrf/sdformat/issues/322. I think this isn't a super important use case for now, so I changed the tests not to have double spaces.